src/main/java/teammates/ui/webapi/output/CourseData.java

```diff
@@ -0,0 +1,39 @@
+package teammates.ui.webapi.output;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+
+/**
+ * The API output format of a course.
+ */
+public class CourseData extends ApiOutput {
```

Comment:
It looks like you haven't modify the frontend to use the Course type?

Reply:
Just to check with you, I'm thinking the following for API output, would they be ok?
DeleteCourse: return Course (only ID is used in frontend)
CreateCourse: return Course
UpdateCourse: return Course
ArchiveCourse: return a new CourseArchiveData object, which contains isArchived, the only variable used in frontend.
SoftDeleteCourse: return Course (only ID is used in frontend)

Reply:
Looks good. `CourseArchiveData` should contain `courseId` and `isArchived`?